### PR TITLE
[SB-285] Create method to fetch all first responders in backend

### PR DESCRIPTION
**Pull request by jennyzeng**
# Description

I need to have a way to fetch the first responders and the current endpoint couldn't meet this need
Also updated old user controller list user method so it doesn't do the field projecting and sorting in memory. Instead, we do them all in query.

# Test Plan

Please explain how this change will be tested.

- [x] Unit test
- [x] Integration test (including unit tests with supertest)
- [ ] Manual test (Please explain below for the test steps)
- [ ] Not Applicable

# Task Link

Please add related jira task links

### CODE REVIEW

**Comment by jennyzeng** [comment]
Current dependencies on/for this PR:
* master
  * **PR #123** <a href="https://app.graphite.dev/github/pr/cmusv-sem/S23SB-SEM-IR/123" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/cmusv-sem/S23SB-SEM-IR/123?utm_source=stack-comment).

File: server/src/controllers/UserController.ts [hunk 1]
```diff
@@ -83,38 +85,117 @@ class UserController {
 
   /**
    * List all users in one's community with their online/office status
+   * This function list all the users in the same community as the current user
```
Comment (bettsongy) [inline server/src/controllers/UserController.ts:88]: This function lists*

File: server/src/controllers/UserController.ts [hunk 2]
```diff
@@ -83,38 +85,117 @@ class UserController {
...
    */
-  async listUsers(role: string) {
-    const Responder = [
-      'Dispatch',
-      'Police',
-      'Fire',
-    ]
-    let users: IUser[] = []
+  async listUsers(user: IUser, communityUserRole: ROLES) {
```
Comment (bettsongy) [inline server/src/controllers/UserController.ts:90]: Is it still necessary to keep the function as async?

**Review by Yqiyuan — APPROVED** [review]

**Review by bettsongy — APPROVED** [review]

File: server/src/controllers/UserController.ts (hunk 1 above)
Comment (jennyzeng) [reply server/src/controllers/UserController.ts:88]: I need to update this in a sequential PR, I will do it in that PR instead.

**Pull request merged**